### PR TITLE
fix(analyzers): preserve space after is in RCS1172 code fix

### LINKED ISSUE #1704: Missing space after "is" in RCS1172 code fix
The code fix for RCS1172 has a missing space after the `is`:

`if (item as MyClass != null)` becomes `if (item isMyClass)`
`if (item as MyClass == null)` becomes `if (!(item isMyClass))`

### PR DESCRIPTION
**Pull request by josefpihrt**
## Summary

- Fix RCS1172 code fix producing `item isMyClass` / `!(item isMyClass)` by preserving whitespace between `is` and the type
- For `== null` checks, build `!(item is MyClass)` without stripping trivia via `WithoutTrivia().Parenthesize()`
- Add test for `item as MyClass == null`

Fixes #1704

## Test plan

- [x] `dotnet test src/Tests/Analyzers.Tests/Analyzers.Tests.csproj --filter "FullyQualifiedName~RCS1172"`

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: src/Analyzers.CodeFixes/CSharp/CodeFixes/UseIsOperatorInsteadOfAsOperatorCodeFixProvider.cs [hunk 1]
```diff
@@ -57,17 +58,35 @@ private static Task<Document> RefactorAsync(
...
+            asExpressionInfo.Type.WithLeadingTrivia(typeLeadingTrivia));
 
         if (nullCheck.IsCheckingNull)
-            newNode = LogicalNotExpression(newNode.WithoutTrivia().Parenthesize()).WithTriviaFrom(newNode);
+        {
+            newNode = LogicalNotExpression(
+                ParenthesizedExpression(
+                    Token(SyntaxTriviaList.Empty, SyntaxKind.OpenParenToken, SyntaxTriviaList.Empty),
+                    newNode,
+                    Token(SyntaxTriviaList.Empty, SyntaxKind.CloseParenToken, SyntaxTriviaList.Empty)));
```
Comment (josefpihrt) [inline src/Analyzers.CodeFixes/CSharp/CodeFixes/UseIsOperatorInsteadOfAsOperatorCodeFixProvider.cs:86]: The old null-check path used `WithTriviaFrom(newNode)` on the `LogicalNotExpression`, which kept leading/trailing trivia from the original comparison. The new `ParenthesizedExpression(...)` construction drops that.

Please restore trivia transfer onto the final expression (at least for the `== null` → `!(... is ...)` path), e.g. `.WithTriviaFrom(node)`, so comments on the replaced condition are not lost.

File: src/Analyzers.CodeFixes/CSharp/CodeFixes/UseIsOperatorInsteadOfAsOperatorCodeFixProvider.cs [hunk 2]
```diff
@@ -57,17 +58,35 @@ private static Task<Document> RefactorAsync(
...
+        }
 
-        newNode = newNode
-            .Parenthesize()
-            .WithFormatterAnnotation();
+        newNode = newNode.WithFormatterAnnotation();
```
Comment (josefpihrt) [inline src/Analyzers.CodeFixes/CSharp/CodeFixes/UseIsOperatorInsteadOfAsOperatorCodeFixProvider.cs]: Nit: the previous code always wrapped the result in `Parenthesize()` before the formatter annotation. Dropping that is probably fine for a full-node replace (and avoids the `WithoutTrivia()` footgun), but please confirm existing `!= null` → `is` tests still format as expected without the extra parentheses.

**Review by josefpihrt — COMMENTED** [review]
Review notes for the RCS1172 `is` spacing fix. The whitespace preservation looks right for #1704; one trivia regression risk.

File: src/Analyzers.CodeFixes/CSharp/CodeFixes/UseIsOperatorInsteadOfAsOperatorCodeFixProvider.cs (hunk 1 above)
Comment (josefpihrt) [reply src/Analyzers.CodeFixes/CSharp/CodeFixes/UseIsOperatorInsteadOfAsOperatorCodeFixProvider.cs:86]: Done. The replacement now uses `WithTriviaFrom(node)` so trailing comments on the original comparison are preserved (`Test_EqualsNull_PreservesTrailingTrivia`).

File: src/Analyzers.CodeFixes/CSharp/CodeFixes/UseIsOperatorInsteadOfAsOperatorCodeFixProvider.cs (hunk 2 above)
Comment (josefpihrt) [reply src/Analyzers.CodeFixes/CSharp/CodeFixes/UseIsOperatorInsteadOfAsOperatorCodeFixProvider.cs]: Confirmed: the existing `!= null` → `item is MyClass` test still formats without extra parentheses, so leaving `Parenthesize()` off is correct for a full-node replace.
